server/game/cards/05-DT/RedhandRegistry.js

```diff
@@ -0,0 +1,29 @@
+const Card = require('../../Card.js');
+
+class RedhandRegistry extends Card {
+    //After A is stolen from you, your opponent skips the “forge a key” step during their next turn.
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                onStealAmber: (event, context) => event.player === context.player
+            },
+            // this looks strange, but we need to create the skip forge event
```

Comment:
I think the problem is that you own the artifact, but it might trigger during your turn or during opponent's turn, so a lastingEffect with a targetComtroller opponent might no be effective always. I wonder if we should pass a "player" to it.

Reply:
Also, the delayed effect might vanish if the artifact is destroyed on the same turn, for example, due a Big Fuse

Reply:
Okay, so looking at this again, I found that there is a untilEndOfMyNextTurn which works perfect!